src/components/sites-sync-modal-selector.tsx

```diff
@@ -0,0 +1,215 @@
+					onRequestClose();
+				} }
+				disabled={ ! selectedSiteId }
+			/>
+		</Modal>
+	);
+}
+
+function SearchSites( {
+	searchQuery,
+	setSearchQuery,
+}: {
+	searchQuery: string;
+	setSearchQuery: ( value: string ) => void;
+} ) {
+	const { __ } = useI18n();
+	return (
+		<div className="flex flex-col px-8 pb-6 border-b border-a8c-gray-5">
+			<SearchControl
+				className="w-full"
+				placeholder={ __( 'Search sites' ) }
+				onChange={ ( value ) => {
+					setSearchQuery( value );
+				} }
+				value={ searchQuery }
+				autoFocus
+			/>
+			<p className="a8c-helper-text text-gray-500">
+				{ __( 'Syncing is supported for sites on the Business plan or above.' ) }
+			</p>
+		</div>
+	);
+}
+
+function ListSites( {
+	syncSites,
+	selectedSiteId,
+	onSelectSite,
+}: {
+	syncSites: SyncSite[];
+	selectedSiteId: null | number;
+	onSelectSite: ( id: number ) => void;
+} ) {
+	return (
+		<div className="flex flex-col overflow-y-auto h-full">
+			{ syncSites.map( ( site ) => (
+				<SiteItem
+					key={ site.id }
+					site={ site }
+					isSelected={ site.id === selectedSiteId }
+					onClick={ () => onSelectSite( site.id ) }
+				/>
+			) ) }
+		</div>
+	);
+}
+
+function SiteItem( {
+	site,
+	isSelected,
+	onClick,
+}: {
+	site: SyncSite;
+	isSelected: boolean;
+	onClick: () => void;
+} ) {
+	const { __ } = useI18n();
+	if ( site.isStaging ) {
+		return null;
+	}
+	const isSyncable = site.syncSupport === 'syncable';
+	const isNeedsTransfer = site.syncSupport === 'needs-transfer';
+	const isUnsupported = site.syncSupport === 'unsupported';
+	return (
+		<div
+			className={ cx(
+				'flex py-3 px-8 items-center border-b border-a8c-gray-0 justify-between',
+				isSelected && 'bg-a8c-blueberry text-white',
+				! isSelected && isSyncable && 'hover:bg-a8c-blueberry-5'
+			) }
+			role={ isSyncable ? 'button' : undefined }
+			onClick={ () => {
+				if ( ! isSyncable ) {
+					return;
+				}
+				onClick();
+			} }
+		>
+			<div className="flex flex-col gap-0.5">
+				<div className={ cx( 'a8c-body', ! isSyncable && 'text-a8c-gray-30' ) }>{ site.name }</div>
+				<div className={ cx( 'a8c-body-small text-a8c-gray-30', isSelected && 'text-white' ) }>
+					{ site.url }
+				</div>
+			</div>
+			{ isSyncable && (
+				<div className="flex gap-2">
+					<Badge
+						className={ cx(
+							isSelected
+								? 'bg-white text-a8c-blueberry text-a8c-blueberry'
+								: 'bg-a8c-green-5 text-a8c-green-80'
+						) }
+					>
+						{ __( 'Production' ) }
+					</Badge>
+					{ site.stagingSiteIds.length > 0 && (
+						<Badge
+							className={ cx( isSelected && 'bg-white text-a8c-blueberry text-a8c-blueberry' ) }
+						>
+							{ __( 'Staging' ) }
+						</Badge>
+					) }
+				</div>
+			) }
+			{ isUnsupported && (
+				<div className="a8c-body-small text-a8c-gray-30"> { __( 'Unsupported plan' ) }</div>
+			) }
+			{ isNeedsTransfer && (
+				<div className="a8c-body-small text-a8c-gray-30">
+					{ __( 'Please enable hosting features' ) }
```

Comment:
This text (and the production + staging labels) should be right aligned.

Let's add some padding between the site name and labels on the right to avoid the overlap.

Reply:
Added right space on https://github.com/Automattic/studio/pull/601/commits/6e057484dc20284a84b0ef1f755f9a817f3bf516

And setting a minimum width solves the other issues.